include/RemotePlugin.h

```diff
@@ -857,12 +889,12 @@ public slots:
 	bool m_failed;
 private:
 	void resizeSharedProcessingMemory();
-
+	bool m_disconnected;
 
 	QProcess m_process;
 	ProcessWatcher m_watcher;
 
-	QMutex m_commMutex;
+	std::recursive_timed_mutex m_commMutex;
```

Comment:
What's the reason you changed to `std::recursive_timed_mutex`? I think we can keep using `QMutex`, avoiding the MinGW compatibility problems.

Reply:
I would like to avoid QT at as much as possible when it's not related to the UI. In the longer term it would, hopefully, make it easier to separate lmms into two parts: a library and what it is now. 

If it's not possible to not use QMutex for this, I'll of course change it.

Reply:
I understand wanting to avoid Qt for core functionality. I'm just not very happy about the MinGW mutex support detection & workaround and would like to avoid it being copy&pasted around the CMake scripts.

What do you think about reverting this PR to `QMutex` for now until (a) we find a cleaner solution or (b) we drop support for old MinGW versions? An example of how (a) could look like can be seen in 57ca68194069c7cd0697a7f707b17b42bb4b648e which is part of #4443. In this commit, I added `include/mingw-std-threads/mutex` which shadows the system `mutex` using `include_directories(BEFORE include/mingw-std-threads)` when it's found necessary. If done correctly, this could fix MinGW thread support for all targets in the project without the need of `#ifdef USE_MINGW_THREADS_REPLACEMENT` checks at every include.

Reply:
I think it's a bad idea, but I won't mind if you change it back to QMutex.

It's a bad idea because if we can't even sever the link between ui and core at this simple level, it will never be done.

Reply:
Give me a couple of days to think this over... I've deleted my previous msg because I responded to quickly; there may be other options.

Are we using mingw for any other reason than to cross-compile to windows?

This pr won't be merged any time soon, so we have all the time we need to find a solution that's clean and acceptable.

Reply:
@justnope Don't worry about using QtCore functionality inside our core, this is completely fine and we'll probably always do it. It doesn't create any coupling between core and UI. The only reason I can see to use `std` mutex here is re-usability outside of LMMS. However, If you ever need to use this code outside of LMMS, you can simply swap `QMutex` for `std::recursive_timed_mutex`, their APIs are compatible.

> Are we using mingw for any other reason than to cross-compile to windows?

We use `winegcc` for `RemoteVstPlugin`, and we support MSYS2 on Windows which uses MinGW.